---
identifier: "CVE-2021-29575"
package_slug: "pypi/tensorflow"
title: "Buffer Overflow"
description: "TensorFlow is an end-to-end open source platform for machine learning.
  The implementation of `tf.raw_ops.ReverseSequence` allows for stack overflow and/or
  `CHECK`-fail based denial of service. The implementation(https://github.com/tensorflow/tensorflow/blob/5b3b071975e01f0d250c928b2a8f901cd53b90a7/tensorflow/core/kernels/reverse_sequence_op.cc#L114-L118)
  fails to validate that `seq_dim` and `batch_dim` arguments are valid. Negative values
  for `seq_dim` can result in stack overflow or `CHECK`-failure, depending on the
  version of Eigen code used to implement the operation. Similar behavior can be exhibited
  by invalid values of `batch_dim`. The fix will be included in TensorFlow We will
  also cherrypick this commit on TensorFlow, TensorFlow, TensorFlow and TensorFlow,
  as these are also affected and still in supported range."
date: "2021-05-18"
pubdate: "2021-05-14"
affected_range: "<=2.1.4||>=2.2.0,<2.2.3||>=2.3.0,<2.3.3||>=2.4.0,<2.4.2"
fixed_versions:
- "2.5.0"
affected_versions: "All versions up to 2.1.4, all versions starting from 2.2.0 before
  2.2.3, all versions starting from 2.3.0 before 2.3.3, all versions starting from
  2.4.0 before 2.4.2"
not_impacted: "All versions after 2.1.4 before 2.2.0, all versions starting from 2.2.3
  before 2.3.0, all versions starting from 2.3.3 before 2.4.0, all versions starting
  from 2.4.2"
solution: "Upgrade to version 2.5.0 or above."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2021-29575"
cvss_v2: "AV:L/AC:L/Au:N/C:N/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H"
uuid: "206f7e2b-b850-4fc0-a882-90ea24ae002e"
